packaging/tags.py

```diff
@@ -225,10 +225,32 @@ def cpython_tags(
                 yield Tag(interpreter, "abi3", platform_)
 
 
-def _generic_abi() -> Iterator[str]:
-    abi = sysconfig.get_config_var("SOABI")
-    if abi:
-        yield _normalize_string(abi)
+def _generic_abi() -> List[str]:
+    """Return the ABI tag based on EXT_SUFFIX"""
+    # ext_suffix is something like this. We want to keep the parts which are
+    # related to the ABI and remove the parts which are related to the
+    # platform:
+    # - linux:   '.cpython-310-x86_64-linux-gnu.so' => cp310
+    # - mac:     '.cpython-310-darwin.so'           => cp310
+    # - win:     '.cp310-win_amd64.pyd'             => cp310
+    # - pypy:    '.pypy38-pp73-x86_64-linux-gnu.so' => pypy38-pp73
+    # - graalpy: '.graalpy-38-native-x86_64-darwin.dylib'
+    #                                               => graalpy-38-native
+
+    ext_suffix = _get_config_var("EXT_SUFFIX", warn=True)
+    assert ext_suffix[0] == "."
+    _, soabi, ext = ext_suffix.split(".")
+    if soabi.startswith("cpython"):
+        abi = "cp" + soabi.split("-")[1]
+    elif soabi.startswith("pypy"):
+        abi = "-".join(soabi.split("-")[:2])
+    elif soabi.startswith("graalpy"):
+        abi = "-".join(soabi.split("-")[:3])
+    elif soabi:
```

Comment:
This does not seem to correctly account for CPython on Windows.

Reply:
correct, thanks. The CPython code always goes through `_cpython_abis()` so that case was not hit. I will add a test.

Reply:
Is this resolved?